resultsdbupdater/utils.py

```diff
@@ -270,6 +270,92 @@ def _test_result_outcome(message):
     return broken_mapping.get(outcome.lower(), outcome)
 
 
+def namespace_from_topic(topic):
+    """
+    Returns namespace from message topic.
+
+    Returns None if the topic does not have the expected format.
+
+    The expected topic format is:
+
+        /topic/VirtualTopic.eng.ci.<namespace>.<artifact>.<event>.{queued,running,complete,error}
+    """
+    if not topic.startswith('/topic/VirtualTopic.eng.ci.'):
+        return None
+
+    topic_components = topic.split('.')
+    if len(topic_components) != 7:
+        return None
+
+    return topic_components[3]
+
+
+def namespace_from_testcase_name(testcase_name):
+    """
+    Returns namespace from test case name.
+
+    Namespace is the component before the first dot.
+    """
+    return testcase_name.split('.', 1)[0]
+
+
+def verify_topic_and_testcase_name(topic, testcase_name):
+    """
+    Verifies that the topic contains same namespace as the test case name.
+
+    If an old topic format is encountered, this test is skipped and a warning
+    is logged. This will be removed in future after everyone uses the new topic
+    format.
+
+    The new topic format is:
+
+        /topic/VirtualTopic.eng.ci.<namespace>.<artifact>.<event>.{queued,running,complete,error}
+
+    Returns true only if topic format is different or the namespace does not
+    match.
+
+    Note: If the "namespace" field in the message contains ".", only the
+    component before the first "." is expected to be in the topic.
+
+    Example:
+
+    - Message body:
+
+        "msg": {
+          "category": "functional",
+          "namespace": "baseos-ci.redhat-module",
+          "type": "tier1",
+          ...
+        }
+
+    - Test case name:
+
+        baseos-ci.redhat-module.tier1.functional
+
+    - Example of expected topic:
+
+        /topic/VirtualTopic.eng.ci.baseos-ci.redhat-module.test.complete
+    """
+    topic_namespace = namespace_from_topic(topic)
+    if not topic_namespace:
+        LOGGER.warn(
+            'The message topic "%s" uses old scheme not containing '
+            'namespace from test case name "%s"',
+            topic, testcase_name)
+        # Old topics are allowed for now.
```

Comment:
How do you know it's an old topic since the function you call also returns `None` if the following is `True`:
```python
if len(topic_components) != 7:
```